frontend/src/post/post-service.ts

```diff
@@ -10,3 +10,7 @@ export const getPosts = async () => {
 export const getPostsByUser = async (username: string) => {
   return (await api.get<Post[]>(`${uri}/by-username/${username}`)).data;
 };
+
+export const deletePost = async (postId: number) => {
+  return (await api.delete<number>(`${uri}/${postId}`)).status;
```

Comment:
Delete post method has an empty response body, so `api.delete<number>` doesn't do anything. Returning a status is unnecessary, see previous comment.